lib/scheduler.js

```diff
@@ -258,8 +258,21 @@ module.exports = class Scheduler {
     const jobName  = job.jobName;
     const time     = job.time;
     const schedule = this._schedules.get(jobName);
-    if (schedule)
-      return schedule._runJob(jobName, time);
+    if (schedule) {
+      // Skip jobs outside of their run window, most commonly
+      // because the scheduled job failed and was returned
+      // to the $schedule queue.
+      const jobTime        = +new Date(time);
+      const nextOccurrence = schedule._getNextSchedule();
+      const lastOccurrence = nextOccurrence - schedule.every;
+      const tooOld         = jobTime < lastOccurrence;
```

Comment:
Say a job is scheduled to run every five minutes. A job is scheduled to run at 9:00. Now it's 9:06, so we should have already scheduled another job, and we don't want to run this one. Aren't we just looking to see if current time > expired time? Expired time being scheduled + every?